authors/apps/friends/views.py

```diff
@@ -0,0 +1,135 @@
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from .models import Friend
+from .serializers import FollowersFollowingSerializer, FollowUnfollowSerializer
+
+
+class FollowUnfollowApiView(generics.RetrieveUpdateDestroyAPIView):
+    """view for following a user"""
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, username, format=None):
+        """
+        Checks if the user associated with the username passed exists
+        If true, check for the user sending the request
+        If same user, returned error message
+        Else create the user relationship
+        Finally return the serialized object of the followed user
+        """
+        try:
+            followed = get_user_model().objects.get(username=username)
+        except ObjectDoesNotExist:
+            raise Http404()
+
+        follower = get_user_model().objects.get(pk=request.user.id)
+
+        if followed == follower:
+            return Response(
+                {"message": "You cannot follow yourself"},
+                status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        if Friend.objects.filter(user_from=follower, user_to=followed):
+            return Response(
+                {"message": "you already follow this user"},
+                status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        Friend.objects.get_or_create(user_from=follower, user_to=followed)
+        user = get_user_model().objects.get(pk=followed.id)
+        serializer = FollowUnfollowSerializer(user)
+        user_id = serializer.data
+        if serializer:
+            return Response(user_id, status=status.HTTP_200_OK)
+
+    def delete(self, request, username, format=None):
+        """
+        Check if the user associated with the username passed exists
+        If true, check for the user sending the request
+        If same user, return an error message
+        Else delete the user relationship
+        Finally the serialized object of the unfollowed user is returned
+        """
+        try:
+            followed = get_user_model().objects.get(username=username)
+        except ObjectDoesNotExist:
+            raise Http404()
+
+        follower = get_user_model().objects.get(pk=request.user.id)
+
+        if followed == follower:
+            return Response(
+                {"message": "You cannot unfollow yourself"},
+                status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        relationship = Friend.objects.filter(
+            user_from=follower, user_to=followed)
+
+        if not relationship:
+            return Response(
+                {"message": "you don't follow this user"},
+                status=status.HTTP_406_NOT_ACCEPTABLE)
+
+        relationship.delete()
+        user = get_user_model().objects.get(pk=followed.id)
+        serializer = FollowUnfollowSerializer(user)
+        user_id = serializer.data
+        if serializer:
```

Comment:
Is there a scenario where serializer will be a falsy value? I feel code is more readable without line 80 and 81.

Reply:
I just realised that is not necessary. Let me get rid of it.